### PR TITLE
Updates to generated filenames

### PR DESCRIPTION
**Pull request by georgedias**
Fixed potential path traversal issues

### CODE REVIEW

**Review by Amndeep7 — DISMISSED** [review]

File: .github/workflows/build-rpm.yml [hunk 1]
```diff
@@ -56,7 +56,7 @@ jobs:
 
       - name: build rpm
         working-directory: ./saf
-        run: rpmbuild -ta "dist/$filename.tar.gz" -D "_rpmdir $(pwd)" -D "version $(cat VERSION)"
+        run: rpmbuild -ta "dist/$filename.tar.gz" -D "_rpmdir $(pwd)" -D "version $(cat VERSION)" --target x86_64
```
[the remark this answers is not in the document]
Comment (georgedias) [reply .github/workflows/build-rpm.yml]: I'm removing after debugging

File: .github/workflows/build-rpm.yml [hunk 2]
```diff
@@ -50,12 +50,23 @@ jobs:
...
         working-directory: ./saf/dist
         run: |
-          # Unpack the original tarball
-          gzip -dc "$(ls)" > "$filename.tar"
-          # Add the spec file into the tarball
-          tar --append --file="$filename.tar" --transform 's|^\.\.|saf|' ../saf.spec
-          # Compress again
-          gzip "$filename.tar"
+          # Find the tar.gz file, use only the first .tar.gz file if multiple files are found
+          original_tarball=$(find . -maxdepth 1 -name "*.tar.gz" | head -n 1)
```
[the remark this answers is not in the document]
Comment (georgedias) [reply .github/workflows/build-rpm.yml]: Code is for debugging and validating/understanding the process

**Review by georgedias — COMMENTED** [review]
fine with me

**Review by georgedias — COMMENTED** [review]
PR should now be ready.

File: src/utils/global.ts [hunk 3]
```diff
@@ -11,40 +11,34 @@ export type SpreadsheetTypes = 'cis' | 'disa' | 'general'
...
-  basename = inputPath.endsWith('\\') ? filePath.at(-2) : filePath.at(-1)
-  if (!basename) {
-    throw new Error('Could not derive basename from file path using \\')
-  }
-
-  return basename
+export function basename(inputPath: string): string {
+  // trim trailing whitespace and path separators ('/'=linux or '\'=windows (note that this could be double backslash on occasion)) from the end of the string
+  const trimmedPath = inputPath.trimEnd().replace(/[\\/]+$/, '')
+
```
[the remark this answers is not in the document]
Comment (Amndeep7) [reply src/utils/global.ts:39]: Added a test and confirmed that this edge case is already handled appropriately and doesn't need the early out that the if statement provides
